audioplayer.cpp

```diff
@@ -0,0 +1,99 @@
+#include "audioplayer.h"
+
+vector<string> getFileNames(const string& folder) {
+    vector<string> names;
+    DIR* directory = opendir(folder.c_str());
+
+    if (directory == nullptr) {
+        abort();
+    }
+
+    struct dirent* ent;
+    while ((ent = readdir(directory)) != nullptr) {
+        if ((strcmp(ent->d_name, "..")) && (strcmp(ent->d_name, "."))) {
+            names.emplace_back(ent->d_name);
+        }
+    }
+    closedir(directory);
+    return names;
+}
+
+void AudioPlayer::closeAudio() {
+    callMciSendString("close mp3");
+}
+
+void AudioPlayer::playAudio() {
+    // save current volume
+    callMciSendString(("setAudio mp3 volume to " + currentVolume));
+
+    // play audio
+    isPlaying = true;
+    callMciSendString("play mp3");
+}
+
+void AudioPlayer::changeVolume(string volume) {
+    string volumeLink;
+
+    currentVolume = volume;
+
+    volumeLink = "setAudio mp3 volume to " + currentVolume;
+    callMciSendString(volumeLink);
+
+    isPlaying = true;
+    callMciSendString("play mp3");
+}
+
+void AudioPlayer::rewindAudio(string trackTiming) {
+    // rewind
+    callMciSendString(("seek mp3 to " + trackTiming + "000"));
+
+    isPlaying = true;
+    callMciSendString("play mp3");
```

Comment:
Reuse `playAudio` method here. Say no to copypaste!

Reply:
done

Reply:
done